src/mlpack/core/optimizers/ada_grad/ada_grad.hpp

```diff
@@ -0,0 +1,164 @@
+   * @param epsilon Value used to initialise the squared gradient parameter.
+   * @param maxIterations Maximum number of iterations allowed (0 means no
+   *        limit).
+   * @param tolerance Maximum absolute tolerance to terminate algorithm.
+   * @param shuffle If true, the function order is shuffled; otherwise, each
+   *        function is visited in linear order.
+   */
+  AdaGrad(DecomposableFunctionType& function,
+      const double stepSize = 0.01,
+      const double epsilon = 1e-8,
+      const size_t maxIterations = 100000,
+      const double tolerance = 1e-5,
+      const bool shuffle = true);
+
+  /**
+   * Optimize the given function using AdaGrad. The given starting point will
+   * be modified to store the finishing point of the algorithm, and the final
+   * objective value is returned.
+   *
+   * @param iterate Starting point (will be modified).
+   * @return Objective value of the final point.
+   */
+  double Optimize(arma::mat& iterate) { return optimizer.Optimize(iterate); } 
+
+  //! Get the instantiated function to be optimized.
+  const DecomposableFunctionType& Function() const { return optimizer.Function(); }
+  //! Modify the instantiated function.
+  DecomposableFunctionType& Function() { return optimizer.Function(); }
+
+  //! Get the step size.
+  double StepSize() const { return optimizer.StepSize(); }
+  //! Modify the step size.
+  double& StepSize() { return optimizer.StepSize(); }
+
+  //! Get the value used to initialise the squared gradient parameter.
+  double Epsilon() const { return adagradUpdate.Epsilon(); }
+  //! Modify the value used to initialise the squared gradient parameter.
+  double& Epsilon() { return adagradUpdate.Epsilon(); }
+
+  //! Get the maximum number of iterations (0 indicates no limit).
+  size_t MaxIterations() const { return optimizer.MaxIterations(); }
+  //! Modify the maximum number of iterations (0 indicates no limit).
+  size_t& MaxIterations() { return optimizer.MaxIterations(); }
+
+  //! Get the tolerance for termination.
+  double Tolerance() const { return optimizer.Tolerance(); }
+  //! Modify the tolerance for termination.
+  double& Tolerance() { return optimizer.Tolerance(); }
+
+  //! Get whether or not the individual functions are shuffled.
+  bool Shuffle() const { return optimizer.Shuffle(); }
+  //! Modify whether or not the individual functions are shuffled.
+  bool& Shuffle() { return optimizer.Shuffle(); }
+
+ private:
+  //! The instantiated function.
+  DecomposableFunctionType& function;
+
+  //! The step size for each example.
+  double stepSize;
+
+  //! The value used to initialise the squared gradient parameter.
+  double epsilon;
+
+  //! The maximum number of allowed iterations.
+  size_t maxIterations;
+
+  //! The tolerance for termination.
+  double tolerance;
+
+  //! Controls whether or not the individual functions are shuffled when
+  //! iterating.
+  bool shuffle;
+
+  //! The AdaGradUpdate policy object.
+  AdaGradUpdate adagradUpdate;
```

Comment:
Yes, I think it would be `optimizer.UpdatePolicy().Epsilon()` but that should work and would be a good way to do it without duplicating any options anywhere.